### DISCUSSION TITLE
Tailwind 3.0: empty tailwind variables get deleted when minifying with vite build, breaking certain styles

### DISCUSSION THREAD

**Discussion by raidsrc** [tailwindlabs/tailwindcss#7000] [Help]
Yo. I want to preface this post by saying that this is my first ever post on anything anywhere on GitHub. I'm a relatively new web-developer-for-fun. I took a class on web development last spring in college for my computer science minor and found it so interesting that I've been doing web development stuff in my free time whenever I can. Tailwind is really fun to use and has made my website building process very enjoyable. 

Anyway, here's the issue: when using Tailwind 3.0, when I build for production using Vite and minify the CSS file, the empty tailwind variables present in my CSS file under the selector `*, ::before, ::after` (such as --tw-blur, --tw-brightness, --tw-contrast, --tw-drop-shadow) all get deleted. The app that I'm building uses a drop shadow effect, and this drop shadow effect appears fine in development when the CSS file is not minified, but breaks in the production build after I've minified. I've confirmed that the minification that `vite build` does is what's deleting the empty tailwind variables. I've also confirmed that, without any of these empty tailwind variables, the drop shadow effect that I desire does not show up. Every single one of the empty tailwind variables need to be present in my minified CSS file in order for the drop shadow effect to work. I've only encountered this problem recently, after upgrading to Tailwind 3.0. I didn't run into this issue when using the previous version of Tailwind. 

How do I minify using `vite build` without breaking the drop shadow effect? I've tried switching between esbuild and terser, and both delete the empty tailwind variables. The only way to get the drop shadow effect to work in production is to disable minification, but that causes my files to become hella large.

Any help is appreciated. Thanks.

**Comment by cossssmin** [comment]
Having the exact same issue, posted here first because I ran into it in Nuxt 2:

https://github.com/nuxt-community/tailwindcss-module/issues/431#issuecomment-1062803204

EDIT: so in case you're using Nuxt, disabling CSS minification in `nuxt.config.js` solves this:

```js
build: {
  html: {
    minify: {
      minifyCSS: false
    }
  }
}
```

Otherwise try to look for what is minifying your CSS and disable it or maybe configure it to not remove empty CSS variables, if possible.
